api/app/src/command/connected-user/get-connected-user.ts

```diff
@@ -34,10 +34,14 @@ export const getProviderDataFromConnectUserOrFail = (
   provider: ProviderOptions
 ) => {
   if (!connectedUser.providerMap) throw new NotFoundError("Could not find provider map");
-  const providerData = connectedUser.providerMap[provider];
-  if (!providerData) throw new NotFoundError("Provider map has no data");
 
-  return providerData;
+  if (connectedUser.providerMap[provider]?.token) {
+    const providerData = connectedUser.providerMap[provider];
+    if (!providerData) throw new NotFoundError("Provider map has no data");
+    return providerData;
+  } else {
+    throw new NotFoundError(`Could not find connect token for ${provider}`);
+  }
```

Comment:
Looking at the code - the only places we use `getProviderDataFromConnectUserOrFail()` is to get the underlying token.

Since this function now checks whether the token exists or not, perhaps we can change this function to be called `getProviderTokenFromConnectedUserOrFail()`, and just return the token itself?

Reply:
I like that!